packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -273,6 +280,13 @@ export const GridRootStyles = styled('div', {
     [`& .${c.columnHeader}:focus, & .${c.cell}:focus`]: {
       outline: `solid ${t.palette.primary.main} 1px`,
     },
+    [`&.${c['root--noToolbar']} [aria-rowindex="1"] [aria-colindex="1"]`]: {
+      borderTopLeftRadius: 'var(--unstable_DataGrid-radius)',
+    },
+    [`&.${c['root--noToolbar']} [aria-rowindex="1"] [aria-colindex="${visibleColumns.length}"]`]: {
```

Comment:
> I try to avoid non-class selectors when possible

How about using `columnHeader--last` instead of `[aria-colindex]`?

```suggestion
    [`&.${c['root--noToolbar']} [aria-rowindex="1"] .${c['columnHeader--last']}`]: {
```

Reply:
I thought about adding a class of that type to avoid JS logic in GridRootStyles but because we don't have a common wrapper for generic cells (including header cells and group header cells) I would have needed to replicate the logic in a few places, I opted for this instead.

Reply:
Wait no this won't work for group headers, their `colcount` is likely smaller than the headers `colcount`. Need to rework this.

Reply:
It's updated, I've added `columnHeader--last` on the groups.